RUFAS/biophysical/manure/storage/open_lot.py

```diff
@@ -0,0 +1,226 @@
+from RUFAS.biophysical.manure.storage.storage import Storage
+from RUFAS.biophysical.manure.storage.storage_cover import StorageCover
+from RUFAS.current_day_conditions import CurrentDayConditions
+from RUFAS.data_structures.animal_to_manure_connection import ManureStream
+from RUFAS.rufas_time import RufasTime
+from RUFAS.units import MeasurementUnits
+
+AMMONIA_EMISSION_COEFFICIENT_IN_OPEN_LOTS: float = 0.36
+"""
+Ammonia emission coefficient used for calculating nitrogen loss in an open lot (unitless).
+"""
+
+NITROUS_OXIDE_COEFFICIENT_IN_OPEN_LOTS: float = 0.02
+"""
+Nitrous oxide coefficient used for calculating nitrogen loss in an open lot (unitless).
+"""
+
+LEACHING_COEFFICIENT: float = 0.035
+"""Leaching coefficient used in the calculation of nitrogen loss in a compost bedded pack barn (unitless)."""
+
+
+class OpenLot(Storage):
+    def __init__(self, name: str, cover: StorageCover, storage_time_period: int | None, surface_area: float):
+        super().__init__(
+            name=name,
+            is_housing_emissions_calculator=False,
+            cover=cover,
+            storage_time_period=storage_time_period,
+            surface_area=surface_area,
+        )
+
+    def process_manure(self, current_day_conditions: CurrentDayConditions, time: RufasTime) -> dict[str, ManureStream]:
+        self._manure_to_process = copy(self._received_manure)
+
+        storage_nitrous_oxide = self._calculate_nitrous_oxide_emissions(
+            NITROUS_OXIDE_COEFFICIENT_IN_OPEN_LOTS, self._manure_to_process.nitrogen
+        )
+        storage_methane = SolidsStorageCalculator.calculate_ifsm_methane_emission(
+            self._manure_to_process.total_volatile_solids, current_day_conditions.mean_air_temperature
+        )
+        storage_nitrogen_leached = SolidsStorageCalculator.calculate_nitrogen_loss_to_leaching(
+            LEACHING_COEFFICIENT, self._manure_to_process.nitrogen
+        )
+        storage_ammonia = self._apply_ammonia_emission(self._manure_to_process.nitrogen)
+
+        self._apply_nitrogen_losses(storage_nitrous_oxide, storage_ammonia, storage_nitrogen_leached)
+
+        carbon_decomposition = SolidsStorageCalculator.calculate_carbon_decomposition(
+            30,
+            self._manure_to_process.non_degradable_volatile_solids,
+            self._manure_to_process.degradable_volatile_solids,
+        )
+        self._apply_dry_matter_loss(storage_methane, carbon_decomposition)
+        self._manure_to_process.volume = self._manure_to_process.mass / ManureConstants.SOLID_MANURE_DENSITY
+
+        received_manure_to_report = copy(self._received_manure)
+        manure_to_return = super().process_manure(current_day_conditions, time)
+
+        self._report_manure_stream(self._stored_manure, "accumulated", time.simulation_day)
+        self._report_manure_stream(received_manure_to_report, "received", time.simulation_day)
+
+        data_origin_name = self.process_manure.__name__
```

Comment:
```suggestion
        data_origin_function = self.process_manure.__name__
```
Or switch the one in composting to `data_origin_name`